plugins/webp-uploads/hooks.php

```diff
@@ -747,13 +747,109 @@ function webp_uploads_render_generator(): void {
 }
 add_action( 'wp_head', 'webp_uploads_render_generator' );
 
+/**
+ * Process a block's content to handle background images for specific block types.
+ *
+ * This function targets blocks like Cover and Group that may use background images,
+ * converting them to modern image formats when appropriate.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-param array{
+ *                    blockName: string|null,
+ *                    attrs: array{
+ *                        id?: positive-int,
+ *                        url?: string,
+ *                        style?: array{
+ *                            background?: array{
+ *                                backgroundImage?: string
+ *                            }
+ *                        }
+ *                    }
+ *                } $block
+ *
+ * @param string|mixed         $block_content The block content.
+ * @param array<string, mixed> $block         The block.
+ * @return string The filtered block content.
+ */
+function webp_uploads_filter_block_background_images( $block_content, array $block ): string {
+	// Because plugins can do bad things.
+	if ( ! is_string( $block_content ) ) {
+		$block_content = '';
+	}
+
+	// Only run on frontend.
+	if ( ! webp_uploads_in_frontend_body() || '' === $block_content ) {
+		return $block_content;
+	}
+
+	$attachment_id = null;
+	$image_url     = '';
+
+	// Process Cover block.
+	if ( 'core/cover' === $block['blockName'] && isset( $block['attrs']['id'] ) ) {
+		$attachment_id = $block['attrs']['id'];
+		$image_url     = $block['attrs']['url'] ?? '';
+	}
+
+	// Process Group block with background image.
+	if ( 'core/group' === $block['blockName'] && isset( $block['attrs']['style']['background']['backgroundImage'] ) ) {
+		$bg_image = $block['attrs']['style']['background']['backgroundImage'];
+		if ( isset( $bg_image['id'] ) ) {
+			$attachment_id = $bg_image['id'];
+			$image_url     = $bg_image['url'] ?? '';
+		}
+	}
+
+	// Abort if there is no associated background image.
+	if ( is_null( $attachment_id ) || '' === $image_url || ! is_array( wp_get_attachment_metadata( $attachment_id ) ) ) {
+		return $block_content;
+	}
```

Comment:
What about this refactor for a bit more hardening?

```suggestion
	$attachment_id = null;
	$image_url     = null;

	if ( 'core/cover' === $block['blockName'] ) {
		if ( isset( $block['attrs']['id'], $block['attrs']['url'] ) ) {
			$attachment_id = $block['attrs']['id'];
			$image_url     = $block['attrs']['url'];
		}
	} elseif ( 'core/group' === $block['blockName'] ) {
		if ( isset( $block['attrs']['style']['background']['backgroundImage']['id'], $block['attrs']['style']['background']['backgroundImage']['url'] ) ) {
			$attachment_id = $block['attrs']['style']['background']['backgroundImage']['id'];
			$image_url     = $block['attrs']['style']['background']['backgroundImage']['url'];
		}
	}

	// Abort if there is no associated background image.
	if (
		! isset( $attachment_id, $image_url ) ||
		$attachment_id <= 0 ||
		'' === $image_url ||
		! is_array( wp_get_attachment_metadata( $attachment_id ) )
	) {
		return $block_content;
	}
```